test/hotspot/jtreg/compiler/loopopts/TestSplitIfPinnedLoadInStripMinedLoop.java

```diff
@@ -138,7 +138,7 @@ int test3(MyClass obj1, MyClass obj2) {
     }
 
     // Same as test2 but with reference to inner loop induction variable 'j' and different order of instructions.
-    // Triggers an assert in PhaseCFG::insert_anti_dependences if loop strip mining verification is disabled:
+    // Triggers an assert in PhaseCFG::raise_above_anti_dependences if loop strip mining verification is disabled:
```

Comment:
Is this test still valid? According to the comment it should trigger an assert but this assert appears to be removed? Is the test correct if the test is passing even though the assert has been removed? See my above comment on the removal of this assertion.

Reply:
Ah, good catch. Let me try to verify that my new asserts also trigger if I revert the fix for [JDK-8260420](https://bugs.openjdk.org/browse/JDK-8260420).

Reply:
Unfortunately, it was not straightforward to revert the fix for [JDK-8260420](https://bugs.openjdk.org/browse/JDK-8260420) (too many changes since then). If `!LCA_orig->dominates(pred_block) || early->dominates(pred_block)` failed at some point, then the new assert `early->dominates(LCA_orig)` must also fail in that situation (in theory). See the details in my other response above.

Reply:
Ok, maybe I was not clear enough.

The comment says:

>     // Triggers an assert in PhaseCFG::raise_above_anti_dependences if loop strip mining verification is disabled:

My question is, does the test on top of which the comment is placed (`test4` right?) ever run with loop strip mining verification disabled and if it does, does the assert get triggered? If `test4` does not do this, seems to me it would be nice to have an additional test that verifies just that rather than accept/assume the comment as valid without a test that actually verifies this. Thoughts?

Reply:
I would assume `test4` is a regression test and that the `assert` no longer triggers in any situation (otherwise we'd still have a bug)? Also, from what I can see, there is no VM flag that disables loop strip mining verification.

Perhaps I'm still misunderstanding you? @TobiHartmann I see you added this test back in 2021, could you help bring us some clarity? This changeset only renames the occurrence of `insert_anti_dependences` in `TestSplitIfPinnedLoadInStripMinedLoop.java` to `raise_above_anti_dependences`.

Reply:
Thanks @TobiHartmann. Note that this changeset does remove the assert, and replaces it with another (stronger, in theory) assert.

Reply:
Ah right, I missed that.

Reply:
@robcasloz suggested that I simply add the new assert to the commit just before the fix which `test4` is a regression test for (much simpler than trying to revert the fix in mainline), and check if it triggers when the old assert triggers. When I manually disable loop strip mining verification (as Tobias suggested), the new assert triggers, as expected, whenever the old assert triggers. Resolving this thread now!